server/game/core/attack/AttackFlow.ts

```diff
@@ -59,22 +65,32 @@ export class AttackFlow extends BaseStepWithPipeline {
             overwhelmDamageOnly = true;
         }
 
-        let damageEvents: GameEvent[];
-
+        const attackerDealsDamageBeforeDefender = this.attack.attackerDealsDamageBeforeDefender();
         if (overwhelmDamageOnly) {
-            damageEvents = [AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).generateEvent(this.attack.target.controller.base, this.context)];
+            const damageEvents = [AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).generateEvent(this.attack.target.controller.base, this.context)];
+            this.context.game.openEventWindow(damageEvents);
+        } else if (attackerDealsDamageBeforeDefender) {
+            const damageEvents = [this.createAttackerDamageEvent()];
+            this.context.game.openEventWindow(damageEvents);
```

Comment:
Events doesn't need to be an array. Could consolidate this into one line as:

```typescript
this.context.game.openEventWindow(this.createAttackerDamageEvent());
```

Ditto for the defender event below at line 77 - 78

Reply:
sent the events seperately